### PR TITLE
Readability and Usability improvements

### PR DESCRIPTION
**Pull request by Baste-RainGames**
tl;dr: made the code easier to read, and removed the need for the prefab

Needing to drop a prefab into the scene is really cumbersome. Unity has a [PostProcessScene] attribute, which adds objects to a scene either when you build the game, or when you start playing the scene in the editor. I made the singleton be generated through that instead of needing the prefab. This makes the Dispatcher significantly more plug-and-play.

I also turned the instance variable private, and made the methods be static. There's no need for a null-check on the instance anymore, as it's always there. This also makes the calling code a bit easier to read.

Finally I turned the _executionQueue into a Queue<IEnumerator>, and made the singleton's Update call StartCoroutine. This avoids a boxing of the IEnumerator into an Action, and looks nicer.

I also did some formatting cleanup to make all code consistent with Pim de Witte's style.

### CODE REVIEW

**Comment by PimDeWitte** [comment]
This is incredible! I'm gonna review, test, and get back to you tomorrow on this. Nice work!
